qiskit/providers/aer/jobs/aerjobset.py

```diff
@@ -218,6 +232,63 @@ def _get_worker_result(self, worker: int, timeout: Optional[float] = None):
                     "Timeout while waiting for JobSet results")
         return result
 
+    def _merge_results(self, result_list: List[Result]):
+        if len(result_list) == 1:
+            return result_list[0]
+
+        master_result = result_list[0].to_dict()
+        sub_result = result_list[1].to_dict()
+        result_list = []
+
+        for (_master_result, _sub_result) in zip(master_result["results"], sub_result["results"]):
+            result_list.append(self._merge_exp(_master_result, _sub_result))
+        master_result["results"] = result_list
+        return Result.from_dict(master_result)
+
+    def _merge_experiments_result(self, results: List[Result]):
```

Comment:
This is a bit confusing with there being two methods for merging results. If one is for combining multiple job results via concatenation, and the other for split circuits into single result via accumulation the names should be clearer, like `_combine_job_results` and `_accumulate_experiment_results` or something like that

Reply:
Change the name of these methods to `_combine_job_results` and `_accumulate_experiment_results`.